### PR TITLE
feat(settings-menu): added Settings Menu

### PR DESCRIPTION
**Pull request by cypscl**
### Tracking Info

Resolves #<issue number>

Make sure your branch name conforms to: `<feature/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

added Settings Menu feature

### Testing (ignore if on ui/ux team)

How did you confirm your changes worked? 

 

### Confirmation of Change (ignore if on ui/ux team)

Upload a screenshot or video (preferable a video), if possible. Otherwise, please provide instructions on how to see the change.

(video on Slack)

### CODE REVIEW

File: source/css/main.css [hunk 1]
```diff
@@ -29,3 +29,8 @@ main {
   color: #fff;
   z-index: 10;
 }
+
+settings-menu {
```
Comment (Kiyo5hi) [inline source/css/main.css]: i think this can be achieved using `:host` pseudo-tag inside the web component declaration

File: source/sort-or-choose.html
Comment (Kiyo5hi) [inline source/sort-or-choose.html:1]: this file should be removed?

File: source/js/translation.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+let wordsList;
+await fetch('./js/wordsList.json')
+  .then((response) => response.json())
+  .then((data) => {
+    wordsList = data;
+  });
+
+export function replaceWords(
```
Comment (Kiyo5hi) [inline source/js/translation.js:16]: maybe JSDoc?

File: source/js/translation.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const translation = dictionary[sourceLanguage][inputString];
+    if (Object.prototype.hasOwnProperty.call(translation, targetLanguage)) {
+      return translation[targetLanguage];
+    }
+  }
+  return inputString;
+}
+
+// Can't translate the output from Fortune Teller
+export function translate(currentLanguage) {
```
Comment (Kiyo5hi) [inline source/js/translation.js:42]: JSDoc +1

File: source/js/translation.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  ) {
+    const translation = dictionary[sourceLanguage][inputString];
+    if (Object.prototype.hasOwnProperty.call(translation, targetLanguage)) {
+      return translation[targetLanguage];
+    }
+  }
+  return inputString;
+}
+
+// Can't translate the output from Fortune Teller
```
Comment (Kiyo5hi) [inline source/js/translation.js:37]: yeah we can leave that for now, maybe do some research to see if you can find a free translate api?

File: source/components/SettingsMenu.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            }
+
+            .slider:hover {
+                opacity: 1;
+            }
+            
+            .slider::-webkit-slider-thumb {
+                -webkit-appearance: none;
+                appearance: none;
+                width: 25px;
```
Comment (Kiyo5hi) [inline source/components/SettingsMenu.js]: stick to `rem` makes the code more consistent

File: source/components/SettingsMenu.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+            <audio id="audioSfx" class="hidden" src="./audio/sfx_harry_potter_expecto_patronum_sound_effect.mp3"></audio>
+
+        `;
+
+    this.shadowRoot.adoptedStyleSheets = [normalize, style];
+
+    document
+      .querySelector('app-header')
+      .shadowRoot.querySelector('#settings > svg')
+      .addEventListener('click', () => {
```
Comment (Kiyo5hi) [inline source/components/SettingsMenu.js:133]: we can leave this for now, or we can emit a custom event in `AppHeader` and this component will listen on that event

**Review by Kiyo5hi — CHANGES_REQUESTED** [review]

File: source/js/translation.js (hunk 4 above)
Comment (cypscl) [reply source/js/translation.js:37]: Shreya said that she'll handle it

**Review by Kiyo5hi — APPROVED** [review]

**Pull request merged**